common-app-lib/src/cmr/common_app/services/kms_lookup.clj

```diff
@@ -280,95 +296,122 @@
-                                    [tm mt-index] (util/time-execution (hash-cache/get-value measurement-cache kms-measurement-cache-key :measurement-name))]
-                                (rl-util/log-redis-read-complete "lookup-by-measurement" kms-measurement-cache-key tm)
-                                mt-index))
-          {:keys [MeasurementContextMedium MeasurementObject MeasurementQuantities]} value
-          measurements (if (seq MeasurementQuantities)
-                         (map (fn [quantity]
-                                {:context-medium MeasurementContextMedium
-                                 :object MeasurementObject
-                                 :quantity (:Value quantity)})
-                              MeasurementQuantities)
-                         [{:context-medium MeasurementContextMedium
-                           :object MeasurementObject}])
-          invalid-measurements (remove #(get measurement-index %) measurements)]
-      (seq invalid-measurements))))
+  (try
+    (when-not (:ignore-kms-keywords context)
+      (let [measurement-cache (hash-cache/context->cache context kms-measurement-cache-key)
+            [tm measurement-index] (util/time-execution (hash-cache/get-value measurement-cache kms-measurement-cache-key :measurement-name))
+            _ (rl-util/log-redis-read-complete "lookup-by-measurement" kms-measurement-cache-key tm)
+            measurement-index (if measurement-index
+                                measurement-index
+                                (let [_ (load-cache-if-necessary context measurement-cache kms-measurement-cache-key)
+                                      [tm mt-index] (util/time-execution (hash-cache/get-value measurement-cache kms-measurement-cache-key :measurement-name))]
+                                  (rl-util/log-redis-read-complete "lookup-by-measurement" kms-measurement-cache-key tm)
+                                  mt-index))
+            {:keys [MeasurementContextMedium MeasurementObject MeasurementQuantities]} value
+            measurements (if (seq MeasurementQuantities)
+                           (map (fn [quantity]
+                                  {:context-medium MeasurementContextMedium
+                                   :object MeasurementObject
+                                   :quantity (:Value quantity)})
+                                MeasurementQuantities)
+                           [{:context-medium MeasurementContextMedium
+                             :object MeasurementObject}])
+            invalid-measurements (remove #(get measurement-index %) measurements)]
+        (seq invalid-measurements)))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "lookup-by-measurement found redis carmine exception. Will return nil result." e)
+          nil)
```

Comment:
Remove the (do and the nil)
(error returns nil.

Please do this in the kms_lookup namespace as well.